### PR TITLE
PMM-6506 More metrics renaming

### PR DESCRIPTION
**Pull request by percona-csalguero**

### CODE REVIEW

File: exporter/metrics.go [hunk 1]
```diff
@@ -271,6 +271,10 @@ func makeMetrics(prefix string, m bson.M, labels map[string]string, compatibleMo
 				continue
 			}
 
+			if renamedMetric := rawToCompatibleRawMetric(rm, specialConversions()); renamedMetric != nil {
```
[the remark this answers is not in the document]
Comment (BupycHuk) [reply exporter/metrics.go:274]: At least `rawToCompatibleRawMetric` should be renamed.

**Review by BupycHuk — CHANGES_REQUESTED** [review]

File: exporter/metrics.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (percona-csalguero) [reply exporter/metrics.go]: Done

File: exporter/v1_compatibility.go [hunk 2]
```diff
@@ -59,10 +54,10 @@ type conversion struct {
 	suffixMapping    map[string]string
 }
 
-func rawToCompatibleRawMetric(rm *rawMetric) *rawMetric {
+func metricRenameAndLabel(rm *rawMetric, convs []conversion) *rawMetric {
```
Comment (BupycHuk) [inline exporter/v1_compatibility.go:57]: it should be moved outside of this package since we use it not only for compatibility mode.

**Review by BupycHuk — CHANGES_REQUESTED** [review]

File: exporter/v1_compatibility.go (hunk 2 above)
Comment (percona-csalguero) [reply exporter/v1_compatibility.go]: Done. Moved to metrics.go

File: exporter/v1_compatibility.go [hunk 3]
```diff
@@ -629,3 +588,65 @@ func walkTo(m primitive.M, path []string) interface{} {
 
 	return val
 }
+
+// specialConversions returns a list of special conversions we want to implement.
+// See: https://jira.percona.com/browse/PMM-6506
+func specialConversions() []conversion {
```
Comment (BupycHuk) [inline exporter/v1_compatibility.go:594]: let's move it too, since it's also not related to compatibility.

**Review by BupycHuk — CHANGES_REQUESTED** [review]

**Review by BupycHuk — APPROVED** [review]

**Review by AlekSi — APPROVED** [review]

**Comment by AlekSi** [comment]
I think we should do the same thing as in rds_exporter where we just dump all metrics in testdata and them compare them: https://github.com/percona/rds_exporter/tree/master/enhanced It will make PRs like that one much more clear – it will be visible in diff what metrics changed

**Pull request merged**